src/containers/AppBody.jsx

```diff
@@ -59,6 +54,29 @@ export class AppBody extends Component {
       cookie.load('orgs').split(' ').forEach(org => dispatch(addOrg(org)));
     }
 
+    this.loadSnippet();
+  }
+
+  componentWillReceiveProps(nextProps) {
+    // Reload snippet if URL changes. Assume it is valid, and
+    // if it is not, the call to loadSnippet will handle it.
+    if (this.state.pathname !== nextProps.router.location.pathname) {
```

Comment:
I tried that, however the router itself appears to mutate its state. E.g. If I go from `/Hopding/blah` to `/`, the value for `router.location.pathname` is `/` in both `this.props` and `nextProps`.